kubernetes/kubernetes_service.go

```diff
@@ -186,6 +186,37 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	return &serviceDetails, nil
 }
 
+func (in *IstioClient) GetDeploymentDetails(namespace string, deploymentName string) (*DeploymentDetails, error) {
+	podsChan, servicesChan := make(chan podsResponse), make(chan servicesResponse)
+
+	deployment, err := in.getDeployment(namespace, deploymentName)
+	if err != nil {
+		return nil, fmt.Errorf("deployment: %s", err.Error())
+	}
+
+	go in.getPods(namespace, deployment.Spec.Selector.MatchLabels, podsChan)
+	go in.getServices(namespace, deployment.Spec.Selector.MatchLabels, servicesChan)
```

Comment:
I think it's not ok. You can fetch pods from deployments selector, but you can't fetch services from deployments selectors.

Actually I can't find a straight way of achieving what we want here. I think we would have to fetch all services within the namespace and, one by one, see if its selector match any of the pods we have. That sounds complicated :-(

But actually, why do we want services in deployment details? If it's for listing source / destination services we'd better look at prometheus metrics instead.

Reply:
oh, I see. I guess that the function name didn't reflect the actual logic. What that function actually does is fetch services by app label. I renamed the method to leverage readability.
Now, and after what we have discussed, I think this meets the requirements.
Anyway, what you proposes it can be handled in another PR as it seems that it can be useful to detect corner cases.